client/flutter/lib/api/linking.dart

```diff
@@ -1,17 +1,21 @@
-import 'package:meta/meta.dart';
-
 class RouteLink {
-  String route;
-  Map<String, String> args;
+  Uri _url;
+
+  String get route => _url.path;
+  Map<String, String> get args => _url.queryParameters ?? {};
+  bool get isExternal {
+    try {
+      return _url.origin != null;
+    } catch (error) {
+      return false;
+    }
+  }
+
+  String get url => _url.toString();
 
-  RouteLink({
-    @required this.route,
-    @required this.args,
-  });
+  RouteLink();
```

Comment:
The intent was that routelink could be made from URIs but fundamentally represent the route link itself.  I think you should preserve the core constructor.